services/UserAuthService.ts

```diff
@@ -42,6 +42,19 @@ export default class UserAuthService {
     });
   }
 
+  public async modifyEmail(user: UserModel, proposedEmail: string): Promise<void> {
+    return this.transactions.readWrite(async (txn) => {
+      const userRepository = Repositories.user(txn);
+
+      await userRepository.upsertUser(user, {
+        email: proposedEmail,
+        state: UserState.PENDING,
+      });
```

Comment:
We need to make sure that the user isn't setting their email to an existing email. Can you add a check here to make sure they aren't setting to a duplicate email?